src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.c

```diff
@@ -712,12 +712,13 @@ mongoc_stream_tls_openssl_new (mongoc_stream_t *base_stream,
 #if OPENSSL_VERSION_NUMBER >= 0x10002000L && !defined(LIBRESSL_VERSION_NUMBER)
    if (!opt->allow_invalid_hostname) {
       struct in_addr addr;
+      struct in6_addr addr6;
       X509_VERIFY_PARAM *param = X509_VERIFY_PARAM_new ();
 
       X509_VERIFY_PARAM_set_hostflags (param,
                                        X509_CHECK_FLAG_NO_PARTIAL_WILDCARDS);
       if (inet_pton (AF_INET, host, &addr) ||
-          inet_pton (AF_INET6, host, &addr)) {
+          inet_pton (AF_INET6, host, &addr6)) {
```

Comment:
Happy to see this was such a straightforward fix. That's not to make any comment on how tricky it might have been to come to this solution :smile:

Reply:
The interesting thing is that this particular change was driven by an ASAN failure that started after enabling certificate verification in the unit tests.  Kevin was actually the one who made the connection between the change I made and the specific JIRA ticket that we had not yet managed to figure out.